components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts

```diff
@@ -65,22 +66,65 @@ export class StatsService {
     return this.httpClient.post<any>(url, body).pipe(
       map(({ controls_summary }) => controls_summary));
   }
-
+  gettimeTrends(days: moment.DurationInputArg1 , intervals: moment.unitOfTime.DurationConstructor , reportQuery: ReportQuery){
+    reportQuery.startDate = moment.utc(reportQuery.endDate).subtract(days , intervals)
+  }
   getNodeTrend(reportQuery: ReportQuery) {
     const url = `${CC_API_URL}/reporting/stats/trend`;
     const interval = 86400;
-
+    let previous_url :string
+    let prev = localStorage.getItem(previous_url);
+    prev.toString();
+    const url_array = ["reporting/nodes" , "reporting/profiles" , "reporting/controls"];
+    var urlcheck = url_array.some(url_var => prev.includes(url_var));
+    if(urlcheck == true){
+      let time_interval = reportQuery.interval;
+      if (time_interval == 0 ){
+        this.gettimeTrends(10 , 'days' , reportQuery)
+      }
+      else if(time_interval == 1){
+        this.gettimeTrends(1 , 'months' , reportQuery)
+      }
+      else if(time_interval == 2) {
+        this.gettimeTrends(3 , 'months' , reportQuery)
+      }
+      else if(time_interval == 3) {
+        this.gettimeTrends(1 , 'years' , reportQuery)
+      }
+      localStorage.setItem(previous_url , url);
+    }
     const formatted = this.formatFilters(reportQuery, false);
     const body = {type: 'nodes', interval, filters: formatted};
-
+  
     return this.httpClient.post<any>(url, body).pipe(
       map(({ trends }) => trends));
   }
 
   getControlsTrend(reportQuery: ReportQuery) {
     const url = `${CC_API_URL}/reporting/stats/trend`;
     const interval = 86400;
-
+    let previous_url : string;
+    let prev = localStorage.getItem(previous_url);
+    prev.toString();
+    const url_array = ["reporting/nodes" , "reporting/profiles" , "reporting/controls"];
+    var urlcheck = url_array.some(url_var => prev.includes(url_var));
+    if(urlcheck == true){
+      let time_interval = reportQuery.interval;
+      if (time_interval == 0 ){
+        this.gettimeTrends(10 , 'days' , reportQuery)
+      }
+      else if(time_interval == 1){
+        this.gettimeTrends(1 , 'months' , reportQuery)
+      }
+      else if(time_interval == 2) {
+        this.gettimeTrends(3 , 'months' , reportQuery)
+      }
+      else if(time_interval == 3) {
+        this.gettimeTrends(1 , 'years' , reportQuery)
+      }
+      localStorage.setItem(previous_url , url);
+    }
```

Comment:
this code is duplicate same as above 
https://github.com/chef/automate/pull/7234/files#diff-cef5eb14d3aea234ffbf728d92d8795c7775087838f5cb5c8e52f51b214adb00R80-R95

you can also create a common function for that to avoid the duplicate code block.